src/txn/txn.c

```diff
@@ -675,87 +652,134 @@ __txn_fixup_prepared_update(WT_SESSION_IMPL *session, WT_TXN_OP *op, WT_CURSOR *
      */
     hs_cbt->compare = 0;
 
-    /* Get current value and convert to full update if it is a modify. */
-    WT_ERR(hs_cursor->get_value(hs_cursor, &hs_stop_ts, &durable_ts, &type_full, hs_value));
+    /* Get current value. */
+    WT_ERR(hs_cursor->get_value(hs_cursor, &hs_stop_durable_ts, &durable_ts, &type_full, hs_value));
+
+    /* The value older than the prepared update in the history store must be a full value. */
+    WT_ASSERT(session, (uint8_t)type_full == WT_UPDATE_STANDARD);
 
     /*
-     * If we found a history value that satisfied the given timestamp, add it to the update list.
```

Comment:
Comment wording can be improved:
"If the history update already has a stop time pair and we are committing the prepared update there is no work to do."

Reply:
done